### DISCUSSION TITLE
Add "Execute to cursor" in VisualFSharp

### DISCUSSION THREAD

**Discussion by ingted** [dotnet/fsharp#16707] [Ideas]
When executing fsx, we usually need to scroll to first line and go back to the last line to execute the selection... Could we add a function to directly execute from first line to the line of the cursor position?

**Comment by ingted** [comment]
Oops, there is a "Run to cursor"...

**Comment by ingted** [comment]
Don't know why, but "Run to cursor" only applies to the f# project (set as startup project) but not a fsx script?!

**Comment by majocha** [comment]
I think run to cursor is a debugger feature in VS.

**Reply by ingted** [reply]
"Debug in Interaction" also applies to VS debugger...

**Comment by smoothdeveloper** [comment]
For evaluation in FSI, I believe a more general concept of extending the selection up and down, to cover top level constructs, through key bindings, followed by the usual "Execute In Interactive" would make more sense than one that re-evaluates always from the top to the cursor.

**Comment by majocha** [comment]
In VS you can do "expand selection to containing block" with `shift + alt + ]`, and then execute.

**Reply by ingted** [reply]
I use VS2022 and in a opened fsx file with cursor in the middle of the file.` shift + alt + ]` doesn't get me a selection from  top level construct neither first line to cursor position... Or I just got invalid perception?

**Reply by majocha** [reply]
Yep, it just selects a block. Ctrl Shift Home selects to the top of everything, I don't think there's an action to select to the top of the block.

**Comment by ingted** [comment]
Ctrl Shift Home is the one!!
